File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/database/mysql/MySQLDatabase.java
package de.jalumu.magma.platform.base.database.mysql;

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;

public class MySQLDatabase {

    private HikariDataSource dataSource;

    public MySQLDatabase(String host, int port, String database, String user, String password) {
        HikariConfig config = new HikariConfig();
        config.setJdbcUrl("jdbc:mysql://" + host + ":" + port + "/" + database);
        config.setUsername(user);
        config.setPassword(password);
        config.setPoolName("MagmaMySQLPool");
        config.addDataSourceProperty("cachePrepStmts", "true");
        config.addDataSourceProperty("prepStmtCacheSize", "250");
        config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");

        dataSource = new HikariDataSource(config);
    }

    public HikariDataSource getDataSource() {
        return dataSource;
    }

    public void shutdown() {
        if (dataSource.isRunning()) {
            dataSource.close();
        }
    }

}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/database/mysql/MySQLDatabase.kt
package de.jalumu.magma.platform.base.database.mysql

import com.zaxxer.hikari.HikariConfig
import com.zaxxer.hikari.HikariDataSource

class MySQLDatabase(host: String, port: Int, database: String, user: String?, password: String?) {
    val dataSource: HikariDataSource

    init {
        val config = HikariConfig()
        config.jdbcUrl = "jdbc:mysql://$host:$port/$database"
        config.username = user
        config.password = password
        config.poolName = "MagmaMySQLPool"
        config.addDataSourceProperty("cachePrepStmts", "true")
        config.addDataSourceProperty("prepStmtCacheSize", "250")
        config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048")
        dataSource = HikariDataSource(config)
    }

    fun shutdown() {
        if (dataSource.isRunning) {
            dataSource.close()
        }
    }
}
